### PR TITLE
Fix `Linting / ESLint (non-excluded files only)` test 

### PR DESCRIPTION
**Pull request by CodeyGuyDylan**
It looks like the `Linting / ESLint (non-excluded files only)` test uses the `prepare` script that was removed in [this PR](https://github.com/Automattic/jetpack/pull/28080). Without it, it is not able to find the `jetpack-image-guide` package.

There may be a better long term fix than just adding it back like this, but this should fix trunk and allow tests to pass at least for now.

#### Changes proposed in this Pull Request:

This PR adds the `prepare` script back to the package.json file for the `jetpack-image-guide` package

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Jetpack product discussion

Context: p1672251936898689/1672246333.206819-slack-CDLH4C1UZ

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:
Changing this file does not run the test that removing `prepare` broke, so I'll provide links to other PRs that are failing this test due to this, as well as link to a branch where I manually added `prepare` back to show that the test passes

1. Go to [this PR](https://github.com/Automattic/jetpack/pull/28095) to see the test failing
image
2. Go to [this PR](https://github.com/Automattic/jetpack/pull/28072), where `prepare` has been added back, to see the test passing
image

### CODE REVIEW

**Comment by chrisbliss18** [comment]
I tested by running `pnpm run lint-required` in the `jetpack` directory. Running it in `trunk` creates the error as shown in the testing instructions. However, switching to the `fix/eslint-test-failure` and running the test again shows the same failure. Unless I'm testing incorrectly, I don't believe that this change resolves the issue.

**Comment by CodeyGuyDylan** [comment]
@chrisbliss18 interesting. Thank you for taking a look at this, this was pretty much uncharted waters for me. On [this PR](https://github.com/Automattic/jetpack/pull/28072), the test failed repeatedly, but as soon as I added the changes in this PR to that branch, it started passing. I'll take a closer using `pnpm run lint-required` like you did. Thanks!

**Comment by chrisbliss18** [comment]
@CodeyGuyDylan I was wrong. I forgot to do something very important when running my tests. After switching to your branch, I forgot to run `jetpack install -r && jetpack install --all` to ensure that all the dependencies were properly updated before running `pnpm run lint-required`.

I just retested your branch by running `jetpack install -r && jetpack install --all` and then `pnpm run lint-required`. With these testing steps, the error no longer appears. I'm going to approve the change. Sorry for forgetting that step and likely sending you down a rabbit hole looking for something else to change.

**Review by chrisbliss18 — APPROVED** [review]
Testing the changes shows that the error is no longer present. Good sleuthing. Thanks for fixing up the issue.

**Comment by CodeyGuyDylan** [comment]
@chrisbliss18 no problem! I'm glad it works now 😄  thanks again for looking at it

**Comment by CodeyGuyDylan** [comment]
Waiting for feedback from the team that needed this script removed before merging, going to see if there's a good solution that solves the test failure problem without breaking what they are working on

Context: p1671775119411749-slack-C01U2KGS2PQ

**Review by haqadn — APPROVED** [review]
It looks like we indeed need it. LGTM.

**Pull request merged**
